diff -uNr --no-dereference buildroot-2022.05/package/Config.in buildroot-2022.05.patched/package/Config.in
--- buildroot-2021.08.1/package/Config.in	2022-06-10 00:43:17.456708191 +0200
+++ buildroot-2021.08.1.patched/package/Config.in	2022-06-10 11:46:43.000000000 +0200
@@ -2448,6 +2448,7 @@
 	source "package/circus/Config.in"
 	source "package/containerd/Config.in"
 	source "package/coreutils/Config.in"
+	source "package/conmon/Config.in"
	source "package/cpulimit/Config.in"
 	source "package/cpuload/Config.in"
 	source "package/daemon/Config.in"
