diff -uNr --no-dereference buildroot-2021.08.1/package/Config.in buildroot-2021.08.1.patched/package/Config.in
--- buildroot-2021.08.1/package/Config.in	2021-10-31 00:51:56.864026605 +0200
+++ buildroot-2021.08.1.patched/package/Config.in	2021-10-31 00:52:52.664434279 +0200
@@ -2450,6 +2450,7 @@
 	source "package/coreutils/Config.in"
 	source "package/conmon/Config.in"
 	source "package/cpuload/Config.in"
+	source "package/cri-tools/Config.in"
 	source "package/daemon/Config.in"
 	source "package/dc3dd/Config.in"
 	source "package/dcron/Config.in"
