diff -ruN --no-dereference buildroot-2021.08.1/package/Config.in buildroot-2021.08.1-a/package/Config.in
--- buildroot-2021.08.1/package/Config.in	2021-10-11 11:46:43.000000000 +0200
+++ buildroot-2021.08.1-a/package/Config.in	2021-11-24 01:37:55.514363631 +0100
@@ -2512,6 +2512,7 @@
 	source "package/scrypt/Config.in"
 	source "package/sdbusplus/Config.in"
 	source "package/seatd/Config.in"
+	source "package/shadow/Config.in"
 	source "package/smack/Config.in"
 	source "package/start-stop-daemon/Config.in"
 	source "package/supervisor/Config.in"
