--- a/loginutils/login.c	2021-03-22 01:34:05.628165667 +0100
+++ b/loginutils/login.c	2021-03-22 01:36:22.719612349 +0100
@@ -434,6 +434,9 @@
 		}
 		/* check that the account is healthy */
 		pamret = pam_acct_mgmt(pamh, 0);
+		if (pamret == PAM_NEW_AUTHTOK_REQD) {
+			pamret = pam_chauthtok (pamh, PAM_CHANGE_EXPIRED_AUTHTOK);
+		}
 		if (pamret != PAM_SUCCESS) {
 			failed_msg = "acct_mgmt";
 			goto pam_auth_failed;
