diff -uNr --no-dereference buildroot-2021.08.1-a/package/Config.in buildroot-2021.08.1-b/package/Config.in
--- buildroot-2021.08.1-a/package/Config.in	2022-01-27 01:29:15.093691336 +0100
+++ buildroot-2021.08.1-b/package/Config.in	2022-01-27 01:29:15.101692056 +0100
@@ -1733,6 +1733,7 @@
 	source "package/c-ares/Config.in"
 	source "package/canfestival/Config.in"
 	source "package/cgic/Config.in"
+	source "package/containernetworking-plugins/Config.in"
 	source "package/cppzmq/Config.in"
 	source "package/curlpp/Config.in"
 	source "package/czmq/Config.in"
diff -uNr --no-dereference buildroot-2021.08.1-a/package/containernetworking-plugins/Config.in buildroot-2021.08.1-b/package/containernetworking-plugins/Config.in
--- buildroot-2021.08.1-a/package/containernetworking-plugins/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2021.08.1-b/package/containernetworking-plugins/Config.in	2020-12-18 00:54:40.618792724 +0100
@@ -0,0 +1,16 @@
+config BR2_PACKAGE_CONTAINERNETWORKING_PLUGINS
+	bool "containernetworking-plugins"
+	depends on BR2_PACKAGE_HOST_GO_TARGET_ARCH_SUPPORTS
+	depends on BR2_PACKAGE_HOST_GO_TARGET_CGO_LINKING_SUPPORTS
+	depends on BR2_TOOLCHAIN_HAS_THREADS
+	depends on BR2_USE_MMU
+
+	help
+	  Some CNI network plugins, maintained by the containernetworking team.
+	  CNI (Container Network Interface), a Cloud Native Computing Foundation
+	  project, consists of a specification and libraries for writing plugins
+	  to configure network interfaces in Linux containers, along with a 
+	  number of supported plugins.
+
+	  https://github.com/containernetworking/plugins
+
diff -uNr --no-dereference buildroot-2021.08.1-a/package/containernetworking-plugins/containernetworking-plugins.mk buildroot-2021.08.1-b/package/containernetworking-plugins/containernetworking-plugins.mk
--- buildroot-2021.08.1-a/package/containernetworking-plugins/containernetworking-plugins.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2021.08.1-b/package/containernetworking-plugins/containernetworking-plugins.mk	2021-11-27 00:26:45.162540125 +0100
@@ -0,0 +1,30 @@
+################################################################################
+#
+# container networking plugins
+#
+################################################################################
+
+CONTAINERNETWORKING_PLUGINS_VERSION = v1.0.1
+CONTAINERNETWORKING_PLUGINS_SITE = https://github.com/containernetworking/plugins/archive/refs/tags
+CONTAINERNETWORKING_PLUGINS_SOURCE = $(CONTAINERNETWORKING_PLUGINS_VERSION).tar.gz
+
+CONTAINERNETWORKING_PLUGINS_LICENSE = Apache-2.0
+CONTAINERNETWORKING_PLUGINS_LICENSE_FILES = LICENSE
+
+CONTAINERNETWORKING_PLUGINS_BUILD_TARGETS = plugins/meta/bandwidth plugins/meta/firewall plugins/meta/portmap plugins/meta/sbr plugins/meta/tuning plugins/meta/vrf
+CONTAINERNETWORKING_PLUGINS_BUILD_TARGETS += plugins/main/bridge plugins/main/host-device plugins/main/ipvlan plugins/main/loopback plugins/main/macvlan plugins/main/ptp plugins/main/vlan
+CONTAINERNETWORKING_PLUGINS_BUILD_TARGETS += plugins/ipam/dhcp plugins/ipam/static plugins/ipam/host-local
+
+CONTAINERNETWORKING_PLUGINS_INSTALL_BINS = $(notdir $(CONTAINERNETWORKING_PLUGINS_BUILD_TARGETS))
+
+define CONTAINERNETWORKING_PLUGINS_POST_INSTALL_HELPER
+	mkdir -p $(TARGET_DIR)/usr/libexec/cni; \
+	for i in $(CONTAINERNETWORKING_PLUGINS_BUILD_TARGETS); do \
+		PLUGIN_FILENAME=$$(basename $$i); \
+		mv $(TARGET_DIR)/usr/bin/$$PLUGIN_FILENAME $(TARGET_DIR)/usr/libexec/cni/$$PLUGIN_FILENAME; \
+	done
+endef
+
+CONTAINERNETWORKING_PLUGINS_POST_INSTALL_TARGET_HOOKS += CONTAINERNETWORKING_PLUGINS_POST_INSTALL_HELPER
+
+$(eval $(golang-package))
