diff -uNr --no-dereference buildroot-2021.08.1-a/package/Config.in buildroot-2021.08.1-b/package/Config.in
--- buildroot-2021.08.1-a/package/Config.in	2021-11-19 00:59:16.595659543 +0100
+++ buildroot-2021.08.1-b/package/Config.in	2021-11-19 00:59:16.603660299 +0100
@@ -2451,6 +2451,7 @@
 	source "package/coreutils/Config.in"
 	source "package/conmon/Config.in"
 	source "package/cpuload/Config.in"
+	source "package/crio/Config.in"
 	source "package/cri-tools/Config.in"
 	source "package/daemon/Config.in"
 	source "package/dc3dd/Config.in"
diff -uNr --no-dereference buildroot-2021.08.1-a/package/crio/Config.in buildroot-2021.08.1-b/package/crio/Config.in
--- buildroot-2021.08.1-a/package/crio/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2021.08.1-b/package/crio/Config.in	2021-11-19 00:41:29.477749085 +0100
@@ -0,0 +1,25 @@
+config BR2_PACKAGE_CRIO
+	bool "cri-o"
+	depends on BR2_PACKAGE_CRUN || BR2_PACKAGE_RUNC
+	depends on BR2_PACKAGE_HOST_GO_TARGET_ARCH_SUPPORTS
+	depends on BR2_PACKAGE_HOST_GO_TARGET_CGO_LINKING_SUPPORTS
+	depends on BR2_TOOLCHAIN_HAS_THREADS
+	depends on !BR2_TOOLCHAIN_USES_UCLIBC
+	depends on BR2_USE_MMU
+	select BR2_PACKAGE_BTRFS_PROGS
+	select BR2_PACKAGE_CONMON
+	select BR2_PACKAGE_LIBGPGME
+	select BR2_PACKAGE_LIBSECCOMP
+	select BR2_PACKAGE_IPROUTE2
+	select BR2_PACKAGE_IPTABLES
+	select BR2_PACKAGE_UTIL_LINUX_NSENTER
+	select BR2_PACKAGE_LVM2
+
+	help
+	  CRI-O is meant to provide an integration path between OCI conformant 
+	  runtimes and the kubelet. Specifically, it implements the Kubelet 
+	  Container Runtime Interface (CRI) using OCI conformant runtimes. 
+	  The scope of CRI-O is tied to the scope of the CRI.
+
+	  https://github.com/cri-o/cri-o
+
diff -uNr --no-dereference buildroot-2021.08.1-a/package/crio/crio.mk buildroot-2021.08.1-b/package/crio/crio.mk
--- buildroot-2021.08.1-a/package/crio/crio.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2021.08.1-b/package/crio/crio.mk	2021-11-11 23:43:11.145089731 +0100
@@ -0,0 +1,45 @@
+################################################################################
+#
+# crio
+#
+################################################################################
+
+CRIO_VERSION = v1.21.4
+CRIO_SITE = $(call github,cri-o,cri-o,$(CRIO_VERSION))
+
+CRIO_LICENSE = Apache-2.0
+CRIO_LICENSE_FILES = LICENSE
+
+CRIO_TAGS = cgo seccomp
+
+CRIO_DEPENDENCIES = libseccomp libgpgme btrfs-progs lvm2
+
+CRIO_BUILD_TARGETS = cmd/crio 
+
+define CRIO_POST_BUILD_PINNS
+	cd $(@D)/pinns; \
+	$(TARGET_MAKE_ENV) $(MAKE) $(TARGET_CONFIGURE_OPTS) $(CRIO_MAKE_OPTS) all
+endef
+
+define CRIO_POST_INSTALL_TARGET_PINNS
+	$(INSTALL) -D -m 0755 $(@D)/bin/pinns $(TARGET_DIR)/usr/bin/pinns
+endef
+
+define CRIO_POST_INSTALL_TARGET_CONTRIBS
+	$(INSTALL) -D -m 0644 $(@D)/contrib/sysconfig/crio $(TARGET_DIR)/etc/sysconfig/crio
+	sed -i -e 's|/usr/local/bin|/usr/bin|' $(@D)/contrib/systemd/crio.service
+	sed -i -e 's|/usr/local/bin|/usr/bin|' $(@D)/contrib/systemd/crio-wipe.service
+	$(INSTALL) -D -m 0644 $(@D)/contrib/systemd/crio.service $(TARGET_DIR)/lib/systemd/system/crio.service
+	$(INSTALL) -D -m 0644 $(@D)/contrib/systemd/crio-shutdown.service $(TARGET_DIR)/lib/systemd/system/crio-shutdown.service
+	$(INSTALL) -D -m 0644 $(@D)/contrib/systemd/crio-wipe.service $(TARGET_DIR)/lib/systemd/system/crio-wipe.service
+endef
+
+CRIO_POST_BUILD_HOOKS += CRIO_POST_BUILD_PINNS
+
+CRIO_POST_INSTALL_TARGET_HOOKS += CRIO_POST_INSTALL_TARGET_PINNS
+CRIO_POST_INSTALL_TARGET_HOOKS += CRIO_POST_INSTALL_TARGET_CONTRIBS
+
+CRIO_INSTALL_BINS = $(notdir $(CRIO_BUILD_TARGETS))
+
+$(eval $(golang-package))
+
