diff -uNr --no-dereference buildroot-2022.05-a/package/Config.in buildroot-2022.05-b/package/Config.in
--- buildroot-2022.05-a/package/Config.in	2022-06-17 01:33:01.493800814 +0200
+++ buildroot-2022.05-b/package/Config.in	2022-06-17 01:33:02.713767765 +0200
@@ -1854,6 +1854,7 @@
 	source "package/librsync/Config.in"
 	source "package/libshairplay/Config.in"
 	source "package/libshout/Config.in"
+	source "package/libslirp/Config.in"
 	source "package/libsocketcan/Config.in"
 	source "package/libsoup/Config.in"
 	source "package/libsrtp/Config.in"
diff -uNr --no-dereference buildroot-2022.05-a/package/libslirp/Config.in buildroot-2022.05-b/package/libslirp/Config.in
--- buildroot-2022.05-a/package/libslirp/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2022.05-b/package/libslirp/Config.in	2021-11-12 00:48:08.308624793 +0100
@@ -0,0 +1,8 @@
+config BR2_PACKAGE_LIBSLIRP
+	bool "libslirp"
+	depends on BR2_USE_MMU
+	select BR2_PACKAGE_LIBGLIB2
+	help
+	  libslirp is a user-mode networking library used by virtual machines, containers or various tools.
+
+	  https://gitlab.freedesktop.org/slirp/libslirp
diff -uNr --no-dereference buildroot-2022.05-a/package/libslirp/libslirp.mk buildroot-2022.05-b/package/libslirp/libslirp.mk
--- buildroot-2022.05-a/package/libslirp/libslirp.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2022.05-b/package/libslirp/libslirp.mk	2022-05-05 16:14:17.219855032 +0200
@@ -0,0 +1,16 @@
+################################################################################
+#
+# libslirp
+#
+################################################################################
+
+LIBSLIRP_VERSION = v4.6.1
+LIBSLIRP_SITE = https://gitlab.freedesktop.org/slirp/libslirp/-/archive/$(LIBSLIRP_VERSION)
+
+LIBSLIRP_LICENSE = 
+LIBSLIRP_LICENSE_FILES = COPYRIGHT
+LIBSLIRP_INSTALL_STAGING = YES
+
+LIBSLIRP_DEPENDENCIES = libglib2
+
+$(eval $(meson-package))
