diff -uNr --no-dereference buildroot-2022.05-a/package/Config.in buildroot-2022.05-b/package/Config.in
--- buildroot-2022.05-a/package/Config.in	2022-06-17 01:33:03.029759205 +0200
+++ buildroot-2022.05-b/package/Config.in	2022-06-17 01:33:03.709740785 +0200
@@ -2360,6 +2360,7 @@
 	source "package/shadowsocks-libev/Config.in"
 	source "package/shairport-sync/Config.in"
 	source "package/shellinabox/Config.in"
+	source "package/slirp4netns/Config.in"
 	source "package/smcroute/Config.in"
 	source "package/sngrep/Config.in"
 	source "package/snmpclitools/Config.in"
diff -uNr --no-dereference buildroot-2022.05-a/package/slirp4netns/Config.in buildroot-2022.05-b/package/slirp4netns/Config.in
--- buildroot-2022.05-a/package/slirp4netns/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2022.05-b/package/slirp4netns/Config.in	2021-11-12 00:37:34.119331969 +0100
@@ -0,0 +1,12 @@
+config BR2_PACKAGE_SLIRP4NETNS
+	bool "slirp4netns"
+	depends on BR2_USE_MMU
+	select BR2_PACKAGE_LIBGLIB2
+	select BR2_PACKAGE_LIBCAP
+	select BR2_PACKAGE_LIBSECCOMP
+	select BR2_PACKAGE_LIBSLIRP
+	help
+	  slirp4netns provides user-mode networking ("slirp") for unprivileged network namespaces.
+
+	  https://github.com/rootless-containers/slirp4netns
+
diff -uNr --no-dereference buildroot-2022.05-a/package/slirp4netns/slirp4netns.mk buildroot-2022.05-b/package/slirp4netns/slirp4netns.mk
--- buildroot-2022.05-a/package/slirp4netns/slirp4netns.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2022.05-b/package/slirp4netns/slirp4netns.mk	2022-05-04 01:02:59.089690214 +0200
@@ -0,0 +1,21 @@
+################################################################################
+#
+# slirp4netns
+#
+################################################################################
+
+SLIRP4NETNS_VERSION = v1.2.0
+SLIRP4NETNS_SITE = $(call github,rootless-containers,slirp4netns,$(SLIRP4NETNS_VERSION))
+
+SLIRP4NETNS_LICENSE = GPL-2.0+
+SLIRP4NETNS_LICENSE_FILES = COPYING
+
+SLIRP4NETNS_DEPENDENCIES = libglib2 libslirp libcap libseccomp
+
+define SLIRP4NETNS_RUN_AUTOGEN
+        cd $(@D) && PATH=$(BR_PATH) ./autogen.sh
+endef
+SLIRP4NETNS_PRE_CONFIGURE_HOOKS += SLIRP4NETNS_RUN_AUTOGEN
+
+
+$(eval $(autotools-package))
