diff -uNr --no-dereference buildroot-2021.08.1-a/package/Config.in buildroot-2021.08.1-b/package/Config.in
--- buildroot-2021.08.1-a/package/Config.in	2022-01-27 01:29:16.661832503 +0100
+++ buildroot-2021.08.1-b/package/Config.in	2022-01-27 01:29:16.669833223 +0100
@@ -1660,6 +1660,7 @@
 	source "package/libxmlpp/Config.in"
 	source "package/libxmlrpc/Config.in"
 	source "package/libxslt/Config.in"
+	source "package/libyajl/Config.in"
 	source "package/libyaml/Config.in"
 	source "package/mxml/Config.in"
 	source "package/pugixml/Config.in"
diff -uNr --no-dereference buildroot-2021.08.1-a/package/libyajl/Config.in buildroot-2021.08.1-b/package/libyajl/Config.in
--- buildroot-2021.08.1-a/package/libyajl/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2021.08.1-b/package/libyajl/Config.in	2021-11-15 23:44:18.004561026 +0100
@@ -0,0 +1,22 @@
+config BR2_PACKAGE_LIBYAJL
+	bool "libyajl"
+	depends on BR2_USE_MMU
+	help
+	  Welcome to Yet Another JSON Library (YAJL)
+	
+          ## Why does the world need another C library for parsing JSON?  
+	
+	  Good question.  In a review of current C JSON parsing libraries I was 
+	  unable to find one that satisfies my requirements.  Those are, 
+	  0. written in C
+	  1. portable
+	  2. robust -- as close to "crash proof" as possible
+	  3. data representation independent
+	  4. fast
+	  5. generates verbose, useful error messages including context of where
+	     the error occurs in the input text.
+	  6. can parse JSON data off a stream, incrementally
+	  7. simple to use
+	  8. tiny
+	
+	  https://github.com/lloyd/yajl
diff -uNr --no-dereference buildroot-2021.08.1-a/package/libyajl/libyajl.mk buildroot-2021.08.1-b/package/libyajl/libyajl.mk
--- buildroot-2021.08.1-a/package/libyajl/libyajl.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2021.08.1-b/package/libyajl/libyajl.mk	2021-10-31 23:20:10.254790898 +0100
@@ -0,0 +1,12 @@
+################################################################################
+#
+# yajl - Yet Another JSON Library
+#
+################################################################################
+
+LIBYAJL_VERSION = 2.1.0
+LIBYAJL_SITE = $(call github,lloyd,yajl,$(LIBYAJL_VERSION))
+LIBYAJL_INSTALL_STAGING = YES
+LIBYAJL_INSTALL_TARGET = YES
+
+$(eval $(cmake-package))
