diff -uNr --no-dereference buildroot-2020.11-a/package/Config.in buildroot-2020.11-b/package/Config.in
--- buildroot-2020.11-a/package/Config.in	2021-02-10 02:25:12.344885740 +0100
+++ buildroot-2020.11-b/package/Config.in	2021-02-10 02:25:12.348886184 +0100
@@ -2423,6 +2423,7 @@
 	source "package/openrc/Config.in"
 	source "package/openvmtools/Config.in"
 	source "package/pamtester/Config.in"
+	source "package/podman/Config.in"
 	source "package/polkit/Config.in"
 	source "package/powerpc-utils/Config.in"
 	source "package/procps-ng/Config.in"
diff -uNr --no-dereference buildroot-2020.11-a/package/podman/Config.in buildroot-2020.11-b/package/podman/Config.in
--- buildroot-2020.11-a/package/podman/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2020.11-b/package/podman/Config.in	2020-12-18 02:03:40.763054949 +0100
@@ -0,0 +1,20 @@
+config BR2_PACKAGE_PODMAN
+	bool "podman"
+	depends on BR2_PACKAGE_HOST_GO_TARGET_ARCH_SUPPORTS
+	depends on BR2_PACKAGE_HOST_GO_TARGET_CGO_LINKING_SUPPORTS
+	depends on BR2_TOOLCHAIN_HAS_THREADS
+	depends on !BR2_TOOLCHAIN_USES_UCLIBC # docker-containerd -> runc
+	depends on BR2_USE_MMU # docker-containerd
+	select BR2_PACKAGE_BTRFS_PROGS
+	select BR2_PACKAGE_CONMON
+	select BR2_PACKAGE_LIBGPGME
+	select BR2_PACKAGE_UTIL_LINUX_NSENTER
+	select BR2_PACKAGE_LIBSECCOMP
+
+	help
+	  Podman (the POD MANager) is a tool for managing containers and images, 
+          volumes mounted into those containers, and pods made from groups of 
+          containers
+
+	  https://github.com/containers/podman
+
diff -uNr --no-dereference buildroot-2020.11-a/package/podman/podman.mk buildroot-2020.11-b/package/podman/podman.mk
--- buildroot-2020.11-a/package/podman/podman.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2020.11-b/package/podman/podman.mk	2020-12-30 01:33:19.486039911 +0100
@@ -0,0 +1,62 @@
+################################################################################
+#
+# podman
+#
+################################################################################
+
+PODMAN_VERSION = v2.2.0
+PODMAN_SITE = $(call github,containers,podman,$(PODMAN_VERSION))
+
+PODMAN_LICENSE = Apache-2.0
+PODMAN_LICENSE_FILES = LICENSE
+
+# other available tags: libdm_no_deferred_remove
+PODMAN_TAGS = cgo exclude_graphdriver_devicemapper
+PODMAN_DEPENDENCIES = libgpgme conmon
+
+# Add btrfs support
+#ifeq ($(BR2_PACKAGE_BTRFS_PROGS),y)
+#PODMAN_DEPENDENCIES += libbtrfs
+#else
+#PODMAN_TAGS += exclude_graphdriver_btrfs
+#endif
+
+# Add seccomp support
+ifeq ($(BR2_PACKAGE_LIBSECCOMP),y)
+PODMAN_TAGS += seccomp
+PODMAN_DEPENDENCIES += libseccomp
+endif
+
+# Add systemd support
+ifeq ($(BR2_PACKAGE_SYSTEMD),y)
+PODMAN_TAGS += systemd
+#PODMAN_DEPENDENCIES += libsystemd
+endif
+
+# Add selinux support
+ifeq ($(BR2_PACKAGE_LIBSELINUX),y)
+PODMAN_TAGS += selinux
+endif
+
+# Add apparmor support
+ifeq ($(BR2_PACKAGE_LIBAPPARMOR),y)
+PODMAN_TAGS += apparmor
+PODMAN_DEPENDENCIES += libapparmor
+endif
+
+PODMAN_GOMOD = .
+PODMAN_BUILD_TARGETS = cmd/podman
+PODMAN_INSTALL_BINS = $(notdir $(PODMAN_BUILD_TARGETS))
+
+# TODO:
+# copy file ./vendor/github.com/containers/common/pkg/seccomp/seccomp.json as /usr/share/containers/seccomp.json
+# generate file /etc/containers/registries.conf
+# wget https://raw.githubusercontent.com/containers/libpod/master/cni/87-podman-bridge.conflist
+# copy file /etc/cni/net.d/87-podman-bridge.conflist
+# generate file /etc/containers/policy.json for example with ./test/policy.json
+# generate file /etc/containers/registries.conf
+# create directory /etc/containers/registries.d
+
+
+$(eval $(golang-package))
+
