diff -uNr a/drivers/usb/host/xhci-rockchip.c b/drivers/usb/host/xhci-rockchip.c
--- a/drivers/usb/host/xhci-rockchip.c	2020-04-13 17:02:18.000000000 +0200
+++ b/drivers/usb/host/xhci-rockchip.c	2021-03-08 16:57:41.235570663 +0100
@@ -31,6 +31,30 @@
 	struct dwc3 *dwc3_reg;
 };
 
+static const char *const speed_names[] = {
+	[DWC3_DCFG_SUPERSPEED] = "UNKNOWN",
+	[DWC3_DCFG_LOWSPEED] = "low-speed",
+	[DWC3_DCFG_FULLSPEED2] = "full-speed",
+	[DWC3_DCFG_HIGHSPEED] = "high-speed",
+	[DWC3_DCFG_SUPERSPEED] = "super-speed"
+};
+
+u32 xhci_usb_get_maximum_speed(struct udevice *dev)
+{
+	const char *maximum_speed;
+	int i;
+
+	maximum_speed = dev_read_string(dev, "maximum-speed");
+	if (maximum_speed == NULL)
+		return DWC3_DCFG_SUPERSPEED;
+
+	for (i = 0; i < ARRAY_SIZE(speed_names); i++)
+		if (strcmp(maximum_speed, speed_names[i]) == 0)
+			return i;
+
+	return DWC3_DCFG_SUPERSPEED;
+}
+
 static int xhci_usb_ofdata_to_platdata(struct udevice *dev)
 {
 	struct rockchip_xhci_platdata *plat = dev_get_platdata(dev);
@@ -89,6 +113,12 @@
 		reg &= ~DWC3_GUSB2PHYCFG_SUSPHY;
 
 	writel(reg, &dwc3_reg->g_usb2phycfg[0]);
+
+	/* Set dwc3 device config register */
+	reg = readl(&dwc3_reg->d_cfg);
+	reg &= ~DWC3_DCFG_SPEED_MASK;
+	reg |= xhci_usb_get_maximum_speed(dev);
+	writel(reg, &dwc3_reg->d_cfg);
 }
 
 static int rockchip_xhci_core_init(struct rockchip_xhci *rkxhci,
@@ -96,17 +126,17 @@
 {
 	int ret;
 
+	rockchip_dwc3_phy_setup(rkxhci->dwc3_reg, dev);
+
+	/* We are hard-coding DWC3 core to Host Mode */
+	dwc3_set_mode(rkxhci->dwc3_reg, DWC3_GCTL_PRTCAP_HOST);
+
 	ret = dwc3_core_init(rkxhci->dwc3_reg);
 	if (ret) {
 		pr_err("failed to initialize core\n");
 		return ret;
 	}
 
-	rockchip_dwc3_phy_setup(rkxhci->dwc3_reg, dev);
-
-	/* We are hard-coding DWC3 core to Host Mode */
-	dwc3_set_mode(rkxhci->dwc3_reg, DWC3_GCTL_PRTCAP_HOST);
-
 	return 0;
 }
 
diff -uNr a/include/linux/usb/dwc3.h b/include/linux/usb/dwc3.h
--- a/include/linux/usb/dwc3.h	2020-04-13 17:02:18.000000000 +0200
+++ b/include/linux/usb/dwc3.h	2021-03-08 17:04:59.383986049 +0100
@@ -204,6 +204,15 @@
 #define DWC3_GTXFIFOSIZ_TXFDEF(n)		((n) & 0xffff)
 #define DWC3_GTXFIFOSIZ_TXFSTADDR(n)		((n) & 0xffff0000)
 
+/* Device Config Register */
+#define DWC3_DCFG_SPEED_MASK			(7 << 0)
+#define DWC3_DCFG_SUPERSPEED			(4 << 0)
+#define DWC3_DCFG_HIGHSPEED			(0 << 0)
+#define DWC3_DCFG_FULLSPEED2			(1 << 0)
+#define DWC3_DCFG_LOWSPEED			(2 << 0)
+#define DWC3_DCFG_FULLSPEED1			(3 << 0)
+
+
 /* Device Control Register */
 #define DWC3_DCTL_RUN_STOP			(1 << 31)
 #define DWC3_DCTL_CSFTRST			(1 << 30)
